app/src/main/java/com/bnyro/clock/domain/model/VibrationPattern.kt

```diff
@@ -0,0 +1,9 @@
+package com.bnyro.clock.domain.model
+
+data class VibrationPattern(
+    val name: String,
+    val pattern: List<Int>,
+    val length: Int = pattern.sum(),
+    val cumulative: List<Int> = pattern.scan(0, Int::plus).drop(1),
+    val fractionalCumulative: List<Float> = cumulative.map { it.toFloat() / length }
+)
```

Comment:
```suggestion
data class VibrationPattern(
    val name: String,
    val pattern: List<Int>
) {
    val length = pattern.sum(),
    val cumulative = pattern.scan(0, Int::plus).drop(1),
    val fractionalCumulative = cumulative.map { it.toFloat() / length }
}
```
These values are never used in the constructors, so it's likely more self-explaining if they're inside the class body.